File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Attribute - Obsolete 01.fs
module Module

let c1: ObsoleteClass = null
let c2 = ObsoleteClass()

let s1: ObsoleteStruct = ObsoleteStruct()
let s2 = ObsoleteStruct()

let i1: IObsoleteInterface = null
let i2 = { new IObsoleteInterface with
             member this.P = 1 }

let d1: ObsoleteDelegate = null
let d2 = ObsoleteDelegate(fun _ -> ())

let e1: ObsoleteEnum = Unchecked.defaultof<_>
let e2 = ObsoleteEnum()
let e3 = ObsoleteEnum.A

Class() |> ignore
Class(1) |> ignore

Class.ObsoleteField |> ignore
Class.ObsoleteMethod()
Class.ObsoleteProperty |> ignore
Class.ObsoleteEvent |> ignore

Enum.A |> ignore

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Attribute - Obsolete 01.cs
using System;

[Obsolete("Class is obsolete")]
public class ObsoleteClass
{
}

[Obsolete("Struct is obsolete")]
public struct ObsoleteStruct
{
}

[Obsolete("Interface is obsolete")]
public interface IObsoleteInterface
{
    int P { get; }
}

[Obsolete("Enum is obsolete")]
public enum ObsoleteEnum
{
    A
}

[Obsolete("Delegate is obsolete")]
public delegate void ObsoleteDelegate();

public class Class
{
    public Class()
    {
    }

    [Obsolete("Constructor is obsolete")]
    public Class(int i)
    {
    }

    [Obsolete("Field is obsolete")] public static readonly int ObsoleteField = 1;

    [Obsolete("Method is obsolete")]
    public static void ObsoleteMethod()
    {
    }

    [Obsolete("Property is obsolete")] public static int ObsoleteProperty => 1;

    [Obsolete("Event is obsolete")] public static event EventHandler ObsoleteEvent;
}

public enum Enum
{
    [Obsolete("Enum field is obsolete")]
    A
}
